integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java

```diff
@@ -74,6 +81,33 @@ public static void init(ContextFactory contextFactory) {
 
   @Override
   public void onOtherEvent(SparkListenerEvent event) {
+    Optional<Long> eventProcessingTimeoutSeconds =
+        OpenLineageSparkListener.getEventProcessingTimeoutSeconds();
+
+    if (!eventProcessingTimeoutSeconds.isPresent()) {
+      log.debug("Processing onOtherEventFn without any timeout");
+      onOtherEventFn(event);
+      return;
+    }
+
+    try {
+      this.callWithTimeout(
+          () -> {
+            log.debug(
+                "Processing onOtherEventFn with timeout: {} seconds",
+                eventProcessingTimeoutSeconds.get());
+            onOtherEventFn(event);
+            return null;
+          },
+          eventProcessingTimeoutSeconds.get(),
+          TimeUnit.SECONDS,
+          true);
+    } catch (Exception e) {
+      log.debug("onOtherEvent Timeout Exception: {}", e.getMessage(), e);
```

Comment:
shouldn't this be `warn` level if not an `error`

Reply:
We had some instances where the timeout logs / error stack trace became huge with warn/errors (timeout for multiple events), as teams usually hide just the debug logs in production. This kinda became an issue for folks trying to debug their actual spark job logs - that's where we had added it as debug